### PR TITLE
Scheduled payment form fields

### PR DESCRIPTION
**Pull request by lukemelia**
This commit has form field layout and input in good shape. It does yet have data wired up such as the correct transaction token list, gas token options, or gas estimation.

image

### CODE REVIEW

File: packages/safe-tools-client/app/components/schedule-payment-form-action-card/index.css [hunk 1]
```diff
@@ -6,6 +6,10 @@
   --boxel-field-label-align: center;
 }
 
+.schedule-payment-form-action-card .boxel-field.schedule-payment-form-action-card__frequency {
+  --boxel-field-label-align: top;
+}
```
Comment (backspace) [inline packages/safe-tools-client/app/components/schedule-payment-form-action-card/index.css:11]: The design has the label centre-aligned:

Schedule Payment - Fill in Info - One Time Payment - MVP - Zeplin 2022-11-10 10-44-39

Cardstack - Percy | Visual testing as a service 2022-11-10 10-43-53

File: packages/safe-tools-client/app/components/schedule-payment-form-action-card/index.gts [hunk 2]
```diff
@@ -29,16 +33,52 @@ export default class SchedulePaymentFormActionCard extends Component<Signature>
...
+    this.paymentDate = this.paymentDate; // trigger reactivity
+  }
+
+  @tracked monthlyUntil: Date | undefined;
+  @action onSetMonthlyUntil(day: Day) {
+    this.monthlyUntil?.setFullYear(day.getFullYear(), day.getMonth(), day.getDate()); 
+    this.monthlyUntil = this.monthlyUntil; // trigger reactivity
+  }
+
+  daysOfMonth = Array.from({length: 28}, (_, i) => i + 1)
```
Comment (backspace) [inline packages/safe-tools-client/app/components/schedule-payment-form-action-card/index.gts:68]: Is this used anywhere?

File: packages/safe-tools-client/app/components/schedule-payment-form-action-card/index.gts [hunk 3]
```diff
@@ -29,16 +33,52 @@ export default class SchedulePaymentFormActionCard extends Component<Signature>
...
+    }
+    if (paymentTypeId === 'monthly') {
+      if (!this.monthlyUntil) {
+        let now = new Date();
+        this.monthlyUntil = new Date(now.getFullYear() + 1, now.getMonth(), now.getDate());
+      }
+      if (!this.paymentDayOfMonth) {
+        this.paymentDayOfMonth = 1;
+      }
+    }
```
Comment (backspace) [inline packages/safe-tools-client/app/components/schedule-payment-form-action-card/index.gts:47]: These seem like things to have assertions on, do you agree? Though maybe it’ll be difficult with ✨ time zones ✨

File: packages/safe-tools-client/app/components/schedule-payment-form-action-card/index.gts [hunk 4]
```diff
@@ -29,16 +33,52 @@ export default class SchedulePaymentFormActionCard extends Component<Signature>
...
     this.selectedPaymentType = paymentTypeId;
   }
 
+  @tracked paymentDate: Date | undefined;
+  @action onSetPaymentDate(day: Day) {
+    this.paymentDate?.setFullYear(day.getFullYear(), day.getMonth(), day.getDate()); 
+    this.paymentDate = this.paymentDate; // trigger reactivity
+  }
```
Comment (backspace) [inline packages/safe-tools-client/app/components/schedule-payment-form-action-card/index.gts:55]: There’s something strange happening for me where the highlighted value is one click behind:

calendar-lag

It’s not happening [here](https://boxel.stack.cards/main/#/docs?s=Components&ss=%3CBoxel%3A%3AInput%3A%3ADate%3E) 😯

**Comment by backspace** [comment]
When does it make sense to start using acceptance tests? I think maybe we could just have one for one-time and one for monthly, just checking what is sent to a mocked SDK function when the button is clicked.

File: packages/safe-tools-client/app/components/schedule-payment-form-action-card/index.gts (hunk 2 above)
Comment (lukemelia) [reply packages/safe-tools-client/app/components/schedule-payment-form-action-card/index.gts:68]: Nope!

**Comment by lukemelia** [comment]
> When does it make sense to start using acceptance tests?

Soon! It would be nice to be able to actually schedule something.

>I think maybe we could just have one for one-time and one for monthly, just checking what is sent to a mocked SDK function when the button is clicked.

I think going for more comprehensive coverage will be worth it.

File: packages/safe-tools-client/app/components/schedule-payment-form-action-card/index.gts (hunk 4 above)
Comment (lukemelia) [reply packages/safe-tools-client/app/components/schedule-payment-form-action-card/index.gts:55]: Fixed.

**Pull request merged**